lnrpc/invoicesrpc/utils.go

```diff
@@ -10,15 +11,42 @@ import (
 	"github.com/lightningnetwork/lnd/zpay32"
 )
 
+// decodePayReq decodes the invoice payment request if present. This is needed,
+// because not all information is stored in dedicated invoice fields. If there
+// is no payment request present, a dummy request will be returned. This can
+// happen with very old invoices and key send.
+func decodePayReq(invoice *channeldb.Invoice,
+	activeNetParams *chaincfg.Params) (*zpay32.Invoice, error) {
+
+	paymentRequest := string(invoice.PaymentRequest)
+	if paymentRequest == "" {
+		preimage := invoice.Terms.PaymentPreimage
```

Comment:
Do we also want to manually set the feature bit we'll use to signal this on the graph level (in the `NodeAnnouncement`)?

Reply:
This would be another way to differentiate this payment from other payments when looking at the set of received payments.

Reply:
My idea was to keep it minimal as long as it is experimental. The node announcement isn't very critical. You can just try to key send and see if it gets pulled.

The payment can be differentiated because there is no pay req, but I think that can also happen with very old payments. It was already an optional field before. Would you want to add a new (experimental) field on the invoices returns by for example `ListInvoices`?

Reply:
The can also be recognized by the custom record that is present and exposed on invoices.

Reply:
The differentation part is solved now with a `is_key_send` rpc bool